src/main/java/com/zaxxer/hikari/pool/HikariPool.java

```diff
@@ -442,7 +442,16 @@ private PoolEntry createPoolEntry()
 
          final long maxLifetime = config.getMaxLifetime();
          if (maxLifetime > 0) {
-            final long variance = maxLifetime > 60_000 ? ThreadLocalRandom.current().nextLong(10_000) : 0;
+        	 long inHours = TimeUnit.MILLISECONDS.toHours(maxLifetime);
+        	 
+        	 // Every hour will add a 10 second random factor for the maxlifetime
+        	 long randomIntervalInMillis = inHours * 10_000;
+        	 
+        	 // Support the up to 10 seconds spread out when the lifetime is less than 1 hour
+        	 if (inHours == 0) {
```

Comment:
@nitincchauhan and @brettwooldridge 
Thanks for the comments.  72 seconds in your example is the max that it can go to and it is a random variance. 

I kind of like @brettwooldridge approach on this to spread this out as much as possible to avoid any system impact.  71 minutes for a day is definitely not bad and we are keeping this spread even during the day to avoid issues.

@brettwooldridge Quick question, let us say, two connections reached the max lifetime and what is the behavior in that time, is the getConnection is going to wait to get those two connections established into the pool?  If that is the case, we will still have the problem if the listeners are taking long time to give out the connections.